core/src/main/java/com/vityuk/ginger/cache/LocalCache.java

```diff
@@ -0,0 +1,51 @@
+/**
+ * com.vityuk.ginger.cache.LocalCache.java
+ *
+ * Copyright (c) 2007-2015 UShareSoft SAS, All rights reserved
+ * @author UShareSoft
+ */
+package com.vityuk.ginger.cache;
+
+import com.vityuk.ginger.util.MiscUtils;
+import com.vityuk.ginger.util.Preconditions;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.concurrent.ExecutionException;
+
+public class LocalCache {
+
+    static class LocalLoadingCache<K, V> extends AbstractLoadingCache<K, V> {
+
+        private final Map<K, ExpireableValue<V>> map = new HashMap<K, ExpireableValue<V>>();
+        private final CacheLoader<? super K, V> cacheLoader;
+        private final long expireAfterWriteNanos;
+
+        public LocalLoadingCache(CacheBuilder<? super K, ? super V> cacheBuilder, CacheLoader<? super K, V>  cacheLoader) {
+            this.expireAfterWriteNanos = cacheBuilder.expireAfterWriteNanos;
+            this.cacheLoader = cacheLoader;
+        }
+
+        @Override
+        public V get(K key) throws ExecutionException {
+            synchronized (map) {
```

Comment:
Yes but in that case we can remove synchronized. There is expired test for removing the association. How do you do that?